src/sysext/sysext.c

```diff
@@ -780,14 +778,23 @@ static int resolve_mutable_directory(const char *hierarchy, char **ret_resolved_
                 return 0;
         }
 
-        path = determine_mutable_directory_path_for_hierarchy(hierarchy);
+        if (arg_mutable == MUTABLE_EPHEMERAL) {
+                root = NULL;
```

Comment:
why is arg_root not supported?

Reply:
Because for ephemeral mode, the overlayfs upperdir and workdir are placed under the sysext workdir, which is hardcoded to be `/run/systemd/sysext`, regardless of `arg_root` (see the invocation of the `merge_subprocess` function). I decided to use sysext workdir, because it is an already existing mountpoint of tmpfs, which seems to be suitable for ephemeral mode.

I'll add a comment here.

Reply:
Silently falling back to changing the rootfs when --root is specified is not ok, please change it so that the incompatible combinations of parameters is rejected and there is a clear error returned

Reply:
Also it needs to be specified in the documentation for the command line options

Reply:
I don't understand - it's not a new issue. The workspace was always defined to be `/run/systemd/sysext`, which is only visible inside the sd-merge subprocess. And this workspace is used for setting up all the extensions and the merged hierarchy before it gets bind mounted to the final place - my PR makes no changes here.

Do you want me to make the workspace to be `arg_root + /run/systemd/sysext` then?

Reply:
The important part is where the overlay is mounted - does this make it so that arg_root is ignored for that?

Reply:
Nothing has changed in this regard, see the filtered `mount` output below (I broke the lines for readability):

Systemd v255.4 (I backported one of the sysext tests to v255.4 tag):
```
sysext on /fake-roots/simple-read-only-with-read-only-hierarchy/usr type overlay
(ro,
 nodev,
 relatime,
 lowerdir=/run/systemd/sysext/meta/usr:
          /run/systemd/sysext/extensions/test-extension/usr:
          /fake-roots/simple-read-only-with-read-only-hierarchy/usr,
 redirect_dir=on,
 nouserxattr)
```

this PR, enabled mutable mode:
```
sysext on /fake-roots/enabled/usr type overlay
(
 rw,
 nodev,
 noatime,
 lowerdir=/run/systemd/sysext/meta/usr:
          /run/systemd/sysext/extensions/test-extension/usr:
          /fake-roots/enabled/usr,
 upperdir=/fake-roots/enabled/var/lib/extensions.mutable/usr,
 workdir=/fake-roots/enabled/var/lib/extensions.mutable/.systemd-usr-workdir,
 redirect_dir=on,
 nouserxattr)
```

this PR, ephemeral mutable mode:
```
sysext on /fake-roots/ephemeral/usr type overlay
(
 rw,
 nodev,
 noatime,
 lowerdir=/run/systemd/sysext/meta/usr:
          /run/systemd/sysext/extensions/test-extension/usr:
          /fake-roots/ephemeral/usr,
 upperdir=/run/systemd/sysext/mh_workspace/usr/usr,
 workdir=/run/systemd/sysext/mh_workspace/usr/.systemd-usr-workdir,
 redirect_dir=on,
 nouserxattr)
```

One interesting thing is that I added in previous PR the `metacopy=off` and it is not displayed here. Or maybe it is translated to `nouserxattr`.

Reply:
ok, that's fine then